### PR TITLE
feature(azure_builder): Azure jenkins builder configuration

### PR DESCRIPTION
**Pull request by fruch**
Adding the code to configure Azure VM agent jenkins pluging currently it's only works with Ubuntu clean image, and not with our SCT-runner image from the gallery

## Testing
- [x] - artifact - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/artifacts-azure-image-test/16/
- [x] - longevity - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/longevity-10gb-3h-azure-test/8/

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by soyacz — APPROVED** [review]
besides few general questions, LGTM

File: sdcm/utils/azure_builder.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+            # Install Docker.
+            apt-get -qq install --no-install-recommends \
+                apt-transport-https ca-certificates curl gnupg-agent software-properties-common
+            curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+            add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+            apt-get -qq install --no-install-recommends docker-ce docker-ce-cli containerd.io
+
+            # Install kubectl.
+            curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/utils/azure_builder.py:92]: this is copy-paste from the code we have for building SCT-runner image.
so the same reason it was needed for sct-runner apply

this is a PR for configuration of Azure, how is GKE related here ? 
i.e. if we have some issue with latest kubectl for GKE, please raise is separately

[the remark this answers is not in the document]
Comment (soyacz) [reply sdcm/utils/azure_builder.py:92]: > this is copy-paste from the code we have for building SCT-runner image. so the same reason it was needed for sct-runner apply
> 
> this is a PR for configuration of Azure, how is GKE related here ? i.e. if we have some issue with latest kubectl for GKE, please raise is separately

True, this was general comment. I created an issue: https://github.com/scylladb/scylla-cluster-tests/issues/5798

**Comment by fruch** [comment]
closing in favor of #5769

**Pull request closed without merging**